project-base/app/tests/FrontendApiBundle/Functional/Cart/MergeCartsTest.php

```diff
@@ -96,6 +96,64 @@ public function testCartIsMergedAfterLogin(): void
         self::assertNull($oldCart);
     }
 
+    public function testCartIsOverwrittenAfterLogin(): void
+    {
+        /** @var \App\Model\Product\Product $anonymouslyAddedProduct */
+        $anonymouslyAddedProduct = $this->getReference(ProductDataFixture::PRODUCT_PREFIX . '5');
+        $anonymouslyAddedProductQuantity = 6;
+        $this->addProductToCustomerCart($anonymouslyAddedProduct, $anonymouslyAddedProductQuantity);
+
+        /** @var \App\Model\Product\Product $anonymouslyAddedProduct2 */
+        $anonymouslyAddedProduct2 = $this->getReference(ProductDataFixture::PRODUCT_PREFIX . '1');
+        $anonymouslyAddedProductQuantity2 = 1;
+        $this->addProductToCustomerCart($anonymouslyAddedProduct2, $anonymouslyAddedProductQuantity2);
+
+        $testCartUuid = CartDataFixture::CART_UUID;
+
+        $loginMutationWithCartUuid = 'mutation {
+                Login(input: {
+                    email: "no-reply@shopsys.com"
+                    password: "user123"
+                    cartUuid: "' . $testCartUuid . '"
+                    shouldOverwriteCustomerUserCart: true
+                }) {
+                    tokens {
+                        accessToken
+                        refreshToken
+                    }
+                    showCartMergeInfo
+                }
+            }
+        ';
+
+        $response = $this->getResponseDataForGraphQlType(
+            $this->getResponseContentForQuery($loginMutationWithCartUuid),
+            'Login',
+        );
+
+        $cart = $this->findCartOfCurrentCustomer();
+
+        self::assertNotNull($cart);
+
+        self::assertFalse($response['showCartMergeInfo']);
+
+        $cartItems = $cart->getItems();
+        self::assertCount(2, $cartItems);
+
+        /** @var \App\Model\Product\Product $firstProduct */
+        $firstProduct = $this->getReference(ProductDataFixture::PRODUCT_PREFIX . '72');
+        self::assertEquals($firstProduct->getFullname(), $cartItems[0]->getName(), 'First product name mismatch');
+        self::assertEquals(2, $cartItems[0]->getQuantity(), 'First product quantity mismatch');
+
+        /** @var \App\Model\Product\Product $secondProduct */
+        $secondProduct = $this->getReference(ProductDataFixture::PRODUCT_PREFIX . '1');
+        self::assertEquals($secondProduct->getFullname(), $cartItems[1]->getName(), 'Second product name mismatch');
+        self::assertEquals(2, $cartItems[1]->getQuantity(), 'Second product quantity mismatch');
```

Comment:
I am confused here - I would expect there will be the two items you have added above - `anonymouslyAddedProduct` with `anonymouslyAddedProductQuantity` and `anonymouslyAddedProduct2` with `anonymouslyAddedProductQuantity2`. What am I missing?

Reply:
Let's see if the new version is better 😆